r1cs-std/src/fields/mod.rs

```diff
@@ -200,6 +200,29 @@ pub trait FieldGadget<F: Field, ConstraintF: Field>:
         Ok(inverse)
     }
 
+    // Returns (self / denominator), but requires fewer constraints than
+    // self * denominator.inverse()
+    // It is up to the caller to ensure that denominator is non-zero,
+    // since in that case the result is unconstrained.
```

Comment:
I know there's a disclaimed in the comment, but if the denominator does end up being zero, then you can set the result to be zero, right? Which is unsound? Is there a way to combine a non-zero check for the denominator with the constraint itself?

```suggestion
    /// Returns (self / denominator), but requires fewer constraints than
    /// self * denominator.inverse()
    /// It is up to the caller to ensure that denominator is non-zero,
    /// since in that case the result is unconstrained.
```

Reply:
Err this comment is wrong, its only unconstrained if the denominator and numerator are zero. (If just the numerator is zero, then result must be 0. If just the denominator is 0, then the constraint can't be satisfied) 

But 0/0 is undefined in general. I know of no way to check that both the numerator and denominator are both non-zero without additional constraints.

Reply:
Hmm so this isn't any different from the `inverse` operation that's already in the system. Ok, looks good me =)